src/base/preferences.h

```diff
@@ -168,26 +171,37 @@ class Preferences: public QObject
     bool isSearchEnabled() const;
     void setSearchEnabled(bool enabled);
 
+    // HTTP Server
     bool isWebUiEnabled() const;
     void setWebUiEnabled(bool enabled);
-    bool isWebUiLocalAuthEnabled() const;
-    void setWebUiLocalAuthEnabled(bool enabled);
     QString getServerDomains() const;
     void setServerDomains(const QString &str);
     quint16 getWebUiPort() const;
     void setWebUiPort(quint16 port);
     bool useUPnPForWebUIPort() const;
     void setUPnPForWebUIPort(bool enabled);
+
+    // Authentication
+    bool isWebUiLocalAuthEnabled() const;
+    void setWebUiLocalAuthEnabled(bool enabled);
+    bool isWebUiAuthIPBypassEnabled() const;
+    void setWebUiAuthIPBypassEnabled(bool enabled);
+    QList<Utils::Net::Subnet> getWebUiAuthBypassSubnets() const;
```

Comment:
I would return subnets as string and convert it when it's really needed. Now it is irrational. In most use cases you convert it back in strings.

Reply:
We're already parsing the user's text input into Subnets to verify if they're valid before saving. The downstream functions that call `getWebUiAuthBypassSubnets` don't always keep them as Subnets, but it makes a lot of sense that the function to get the subnets would return items of type Subnet.

Reply:
>but it makes a lot of sense that the function to get the subnets would return items of type Subnet.

What's the point? I don't see any sense. After all, you even do not check data loaded from config. Do not forget that it can be changed manually.

Reply:
> What's the point? I don't see any sense.

The code is self documenting. The return type matches what you'd expect based on the name of the function (ie. a function to get a list of subnets would return a list of objects of type Subnet).

> After all, you even do not check data loaded from config. Do not forget that it can be changed manually.

Subnet whitelist entries are checked before they're saved. A user can circumvent the process by editing the config manually, but who's going to do this? And why do we care? I really don't see how this point is relevant.

Reply:
>who's going to do this? And why do we care?

The manual config editing is the only way when qBittorrent built without GUI. And Yes, we should care about it just as you take care of the data entered via the appropriate dialog in GUI.

Reply:
Unfortunately, it may be absent in most legacy code. But lately I constantly remind to all contributors that we should check the input data from any external sources.

Reply:
Done.

Reply:
FOA, thx a lot for this functionality @Piccirello . Beyond all input validations and formatting, what will the end entry in the preferences file to define the pre-authenticated subnet? Something like 192.168.0.0/24 or anything different?

Reply:
Each entered subnet is "normalized" by passing it through `QHostAddress::parseSubnet()`. Then the entire list (excluding any invalid subnets) is saved to the preferences file.

For example, my whitelist contains these entries:
```
192.168.10/24
192.168.50/24
```

and my `qBittorrent.conf` contains this line:
```
WebUI\AuthSubnetWhitelist=192.168.10.0/24, 192.168.50.0/24
```

Reply:
Perfect @Piccirello. Thx again for this feature. As soon as it's accepted I'll test it against the altWebUI stuff recently implemented.